Bug: MSW mock handler imports aren't included

When generating mock handlers not all schemas are imported.

```
 % ./build.sh 
🍻 Start orval v7.9.0 - A swagger client generator for typescript
🎉 operations - Your OpenAPI spec has been converted into ready to use orval!
generated/v1_operations.msw.ts:67:76 - error TS2304: Cannot find name 'Status'.

67 export const getOperationsCancelOperationMockHandler = (overrideResponse?: Status | ((info: Parameters<Parameters<typeof http.post>[1]>[0]) => Promise<Status> | Status)) => {
                                                                              ~~~~~~

generated/v1_operations.msw.ts:67:152 - error TS2304: Cannot find name 'Status'.

67 export const getOperationsCancelOperationMockHandler = (overrideResponse?: Status | ((info: Parameters<Parameters<typeof http.post>[1]>[0]) => Promise<Status> | Status)) => {
                                                                                                                                                          ~~~~~~

generated/v1_operations.msw.ts:67:162 - error TS2304: Cannot find name 'Status'.

67 export const getOperationsCancelOperationMockHandler = (overrideResponse?: Status | ((info: Parameters<Parameters<typeof http.post>[1]>[0]) => Promise<Status> | Status)) => {
                                                                                                                                                                    ~~~~~
```

Repository for reproduction: https://github.com/bweston92/orval-mock-import-missing

Generated imports for client:

https://github.com/bweston92/orval-mock-import-missing/blob/30045ca73ff2fa3b96a01fd6a9fd05a1a4dcb2bf/generated/v1_operations.ts#L36-L43

```typescript
import type {
  CancelOperationRequest,
  CosmicapisProtoEmpty,
  ListOperationsResponse,
  Operation,
  OperationsListOperationsParams,
  Status
} from './v1_operations.schemas';
```

Generated imports for mock handlers: 

https://github.com/bweston92/orval-mock-import-missing/blob/30045ca73ff2fa3b96a01fd6a9fd05a1a4dcb2bf/generated/v1_operations.msw.ts#L17-L21

```typescript
import type {
  CosmicapisProtoEmpty,
  ListOperationsResponse,
  Operation
} from './v1_operations.schemas';
```

... causing the following definition to fail ...

```typescript
export const getOperationsCancelOperationMockHandler = (overrideResponse?: Status | ((info: Parameters<Parameters<typeof http.post>[1]>[0]) => Promise<Status> | Status)) => {
  return http.post('http://localhost:3333/operations/v1/operations/:operation:cancel', async (info) => {await delay(0);
  if (typeof overrideResponse === 'function') {await overrideResponse(info); }
    return new HttpResponse(null,
      { status: 200,
        
      })
  })
}
```

--- a/packages/mock/src/msw/index.ts
+++ b/packages/mock/src/msw/index.ts
@@ -155,19 +155,18 @@ export const ${handlerName} = (overrideResponse?: ${returnType} | ((${infoParam}
   })
 }\n`;
 
-  const includeResponseImports =
-    isReturnHttpResponse && !isTextPlain
-      ? [
-          ...imports,
-          ...response.imports.filter((r) => {
-            // Only include imports which are actually used in mock.
-            const reg = new RegExp(`\\b${r.name}\\b`);
-            return (
-              reg.test(handlerImplementation) || reg.test(mockImplementation)
-            );
-          }),
-        ]
-      : imports;
+  const includeResponseImports = !isTextPlain
+    ? [
+        ...imports,
+        ...response.imports.filter((r) => {
+          // Only include imports which are actually used in mock.
+          const reg = new RegExp(`\\b${r.name}\\b`);
+          return (
+            reg.test(handlerImplementation) || reg.test(mockImplementation)
+          );
+        }),
+      ]
+    : imports;
 
   return {
     implementation: {

--- a/tests/configs/react-query.config.ts
+++ b/tests/configs/react-query.config.ts
@@ -13,6 +13,23 @@ export default defineConfig({
       target: '../specifications/petstore.yaml',
     },
   },
+  noContentWithDefault: {
+    output: {
+      target: '../generated/react-query/no-content-with-default/endpoints.ts',
+      schemas: '../generated/react-query/no-content-with-default/model',
+      mode: 'split',
+      client: 'react-query',
+      mock: {
+        type: 'msw',
+        delay: 0,
+        useExamples: true,
+      },
+      headers: true,
+    },
+    input: {
+      target: '../specifications/no-content-with-default.yaml',
+    },
+  },
   petstoreTagsSplit: {
     output: {
       target: '../generated/react-query/petstore-tags-split/endpoints.ts',

--- a/tests/specifications/no-content-with-default.yaml
+++ b/tests/specifications/no-content-with-default.yaml
@@ -1,0 +1,28 @@
+openapi: '3.0.0'
+info:
+  version: 1.0.0
+  title: Swagger Errors
+  license:
+    name: MIT
+paths:
+  /error:
+    get:
+      summary: Get an Error Item
+      operationId: createItems
+      responses:
+        '200':
+          description: Success
+          content: {}
+        default:
+          description: Error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+components:
+  schemas:
+    ErrorResponse:
+      type: object
+      properties:
+        prop:
+          type: string

